### DISCUSSION TITLE
Data_Importer: 401: Unauthenticated despite good token

### DISCUSSION THREAD

**Discussion by chevdor** [firefly-iii/firefly-iii#6263] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely me, not you](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I have found similar issues such as #5580 but the threads don't help me solve the issue.

I upgraded from Firefly 5.6.5 to 5.7.10.
The upgrade went fine and after the migration was done, firefly is working fine.

I used to use the csv-importer successuffly and now switched to the data-importer.
I don't remember running into issues with the csv-importer.

After getting the first 401 issues, I decided to remove my old token and create new ones.
It did not help. I did try clearning my cookies with no success.

```
Error accessing "http://192.168.0.4:7041/api/v1/about?". Status code is 401. Body is: {"message":"Unauthenticated","exception":"AuthenticationException"}
```

### Expected behaviour

The auth works

### Debug information

Debug information generated at 2022-07-23 18:46:48 Europe/Berlin for Firefly III version **5.7.10**.

| Scope | Version |
| --- | --- |
| Firefly III | 5.7.10 |
| Firefly III API | 1.5.6 |
| Build | 222, 21-07-2022 17:05:02 CEST |
| DB version | 18 (exp. 18) |
| PHP | `8.1.8` |
| Host | `Linux` |

| System info | Value |
| --- | --- |
| System TZ | Europe/Berlin |
| Browser TZ | Europe/Berlin |
| App environment | local |
| App debug mode | false |
| App cache driver | redis |
| App logging | info, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | apache2handler |
| Default language | en_US |
| Default locale | equal |
| BCscale | 24 |
| DB drivers | mysql, pgsql, sqlite |
| Current driver | pgsql |
| Trusted proxies (.env) | `` |

| User info | Value |
| --- | --- |
| Login provider | eloquent  |
| User guard | web  |
| Headers | N/A, N/A  |
| Stateful domains |  |
| Session start |  |
| Session end |  |
| Session first |  |
| User ID | 1 |
| User language | en_US |
| User locale | en_US |
| Attempt at "en_US.utf8" | 'en_US.utf8' |
| Attempt at "en_US.UTF-8" | 'en_US.UTF-8' |
| User agent | Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.5060.114 Safari/537.36 |

### Steps to reproduce

- start firefly (docker)
- start the data-importer container 

### Additional info

I am using Firefly for a while and went thru several upgrades so there may be remains...
I also noticed that unlike what the doc mentioned, artisan is not located under `/var/www/firefly` but under `/var/www/html`.
That's probably due to the docker image being slightly different.

**Comment by JC5** [comment]
I can't do a lot about this, the code that manages the authentication is pretty foolproof.

But since you're using the UI, make sure you press "Reauthenticate" a few times. The authentication data is stored in your cookies as well.

**Comment by chevdor** [comment]
Yes, I tried that. I tried also "incognito" (Brave) and also Safari to exclude browser issues.
The outcome remains the same.

**Reply by JC5** [reply]
Sorry, in that case I really need you to double check how you copy and paste these tokens. You can also try them on the public version: https://data-importer.firefly-iii.org/token

- No newlines
- No quotes
- Try setting just the ID and the URL, skip the token.

**Comment by chevdor** [comment]
Yes, I considered that as well and re-did it carefully, even going thru a simple editor to ensure the copy/paster did not sneak in new lines in or extra spaces. It looks clean.

ok, so if I provide no token, it will be asked. I will test that and report back.

**Comment by chevdor** [comment]
Problem solved !

I found the issue and it is on my end.
I am using Portainer and this thing does NOT edit the ENV as it should (it confirms it does but does not).
So I **thought** I had the right (new) token but it was still using the old revoked one, so it would make sense for the importer to complain.

This time with the proper token in branch new container and a `Reauthenticate`, it went through all fine.

Thansk for the troubleshooting @JC5

**Reply by JC5** [reply]
Thanks for letting me know what the issue was! I'll remember it for the next one.

**Reply by JC5** [reply]
Thanks for letting me know what the issue was! I'll remember it for the next one.
